HIP/hip-1069.md

```diff
@@ -0,0 +1,91 @@
+
+## Motivation
+The Hedera SDK allows setting KeyLists for token keys, which is a useful feature for multi-signature setups and threshold-based authorizations. However, the HTS precompile function `updateTokenKeys` does not currently support KeyLists, making it impossible to update token keys with multiple signatories directly on-chain. This limitation creates a gap in functionality, especially for projects that rely heavily on multi-signature control within contract environments, such as Axelar's cross-chain integrations.
+
+## Rationale
+Supporting KeyLists directly within the `updateTokenKeys` precompile aligns with Hedera’s goal of providing flexible, decentralized token management. By allowing contract-based applications to set KeyLists, developers can enhance security and improve token access controls within contract-managed tokens, without needing to rely on external SDK calls or workarounds.
+
+## User Stories
+1. **As a developer** managing a cross-chain bridge, I want to use KeyLists to control token keys, allowing multiple signatories to authorize key updates through HTS precompiles.
+2. **As a DApp creator**, I want to assign a `KeyList` as the supply key for my token, using a threshold mechanism to ensure decentralized control over supply adjustments.
+3. **As a DAO member**, I want to update token keys on-chain using a `KeyList` to allow multi-signature voting for key management.
+
+## Specification
+The following update is proposed to the HTS `updateTokenKeys` precompile function:
+
+### Updated `updateTokenKeys` Precompile Function
+Add support for `KeyList` types within `IHederaTokenService.TokenKey[]` in the `updateTokenKeys` function. The following example demonstrates how a KeyList parameter should be accepted by the precompile:
+
+```solidity
+/// Operation to update token keys with KeyList support
+/// @param token The token address
+/// @param keys The token keys, including KeyLists for multi-key support
+/// @return responseCode The response code for the status of the request. SUCCESS is 22.
+function updateTokenKeys(
+    address token,
+    IHederaTokenService.TokenKey[] memory keys
+) internal returns (int64 responseCode) {
+    (bool success, bytes memory result) = precompileAddress.call(
+        abi.encodeWithSelector(
+            IHederaTokenService.updateTokenKeys.selector,
+            token,
+            keys
+        )
+    );
+    (responseCode) = success
+        ? abi.decode(result, (int32))
+        : HederaResponseCodes.UNKNOWN;
+}
+
+Example Usage: 
+
+IHederaTokenService.TokenKey memory key = IHederaTokenService.TokenKey({
+    keyType: KeyType.SUPPLY,
+    key: KeyList({
```

Comment:
Q:  Are we aiming to support both KeyList and Threshold key types? I can see only KeyList mentioned but this example is using a Threshold type for the keyless.